machine.go

```diff
@@ -554,11 +578,15 @@ func (machine Machine) toNode(
 		keyExpiry = time.Time{}
 	}
 
+	name := machine.Name
+	if len(machine.Nickname) > 0 {
+		name = machine.Nickname
+	}
```

Comment:
I think we need to think about this a bit differently, if we implement this feature, we will have two name fields:

- `Name`
- `Nickname`

`Name` should be the _exact_ hostname that is passed by the Tailscale client, and should _not_ be used in the DNS config.

`Nickname` should be set as the Normalized name of `Name` on registration (e.g. it will always have a value) and then it _can_ be changed by the user later.

`Nickname` should therefore always be used for the DNS Config.

Reply:
Then, we have to store the state of "[Auto-generate from OS hostname](https://tailscale.com/kb/1098/machine-names/#renaming-a-machine)" checkbox in database, right? Otherwise, each client update will overwrite this field.

Reply:
Yes, so the flow should be:

1. Machine sends a register request with hostname: TailScaleNode-123
2. We set the `HostName` (currently `Name`) field to TailScaleNode-123
3. We Normalize TailScaleNode-123 and set the result to `GivenName` (currently `Nickname`)

Then if a user wants to _rename_ the host, we let them do that and use the validation part of the Normalization infra.

Reply:
It does look like the `GivenName` should be unique?

Reply:
Yes for the DNS to work properly, normalization function should check if GivenName is already registered in DB and add something to it. This step should behave proerly automatically without the use of an admin or the user to change anything. If the user is not satisfied with the name he could change it manually on the tailscale node.